src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -158,17 +178,31 @@ public bool TryGetWellKnownAttribute(WellKnownAttributeFlags flag, out bool valu
         /// <param name="handle">The property parameter doesn't have a name in metadata,
         /// so this is the handle of a corresponding accessor parameter, if there is one,
         /// or of the ParamInfo passed in, otherwise).</param>
+        /// <param name="parameterInfo" />
         /// <param name="isBad" />
-        /// <param name="parameter"></param>
         internal static PEParameterSymbol Create(
             PEModuleSymbol moduleSymbol,
             PEPropertySymbol containingSymbol,
             int ordinal,
             ParameterHandle handle,
-            ParamInfo<TypeSymbol> parameter,
+            ParamInfo<TypeSymbol> parameterInfo,
             out bool isBad)
         {
-            return Create(moduleSymbol, containingSymbol, ordinal, parameter.IsByRef, parameter.RefCustomModifiers, parameter.Type, handle, parameter.CustomModifiers, out isBad);
+            var parameter = Create(moduleSymbol, containingSymbol, ordinal, parameterInfo.IsByRef, parameterInfo.RefCustomModifiers, parameterInfo.Type, handle, parameterInfo.CustomModifiers, out isBad);
+            var inAttributeModreq = parameter.RefCustomModifiers.Any(modifier => !modifier.IsOptional && modifier.Modifier.IsWellKnownTypeInAttribute());
+
+            if (parameter.RefKind == RefKind.In)
+            {
+                // An in parameter should not have this modreq, unless the method was virtual or abstract.
+                isBad |= (containingSymbol.IsVirtual || containingSymbol.IsAbstract) != inAttributeModreq;
```

Comment:
>containingSymbol.IsVirtual || containingSymbol.IsAbstract [](start = 26, length = 57)

It looks like we should also check IsOverride

Reply:
AFAIU, this is checked when we load the override that it matches the overridden correctly, so we don't need that extra check here.

---
In reply to: [166439052](https://github.com/dotnet/roslyn/pull/24524#discussion_r166439052) [](ancestors = 166439052)